### ISSUE TITLE
SSLEOFError: EOF occurred in violation of protocol

### ISSUE THREAD

**Issue by alessiodionisi** [ajenti/ajenti#728]
I tried

```
sudo easy_install -U gevent
```

But did not solve the problem...

```
2015-05-27 15:09:34,244 INFO     users.check_password(): user root logged in through AjentiSyncProvider from 2.239.61.150
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/gevent-1.0.2-py2.7-linux-x86_64.egg/gevent/greenlet.py", line 327, in run
    result = self._run(*self.args, **self.kwargs)
  File "/usr/local/lib/python2.7/dist-packages/gevent-1.0.2-py2.7-linux-x86_64.egg/gevent/server.py", line 102, in wrap_socket_and_handle
    ssl_socket = self.wrap_socket(client_socket, **self.ssl_args)
  File "/usr/local/lib/python2.7/dist-packages/gevent-1.0.2-py2.7-linux-x86_64.egg/gevent/_sslgte279.py", line 636, in wrap_socket
    ciphers=ciphers)
  File "/usr/local/lib/python2.7/dist-packages/gevent-1.0.2-py2.7-linux-x86_64.egg/gevent/_sslgte279.py", line 248, in __init__
    raise x
SSLEOFError: EOF occurred in violation of protocol (_ssl.c:590)
<Greenlet at 0x7fa4ef860cd0: <bound method SocketIOServer.wrap_socket_and_handle of <SocketIOServer at 0x7fa4ef9e9590 fileno=8 address=0.0.0.0:8000>>(<socket at 0x7fa4ee41be10 fileno=17 sock=172.31.16, ('2.239.61.150', 38716))> failed with SSLEOFError

Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/gevent-1.0.2-py2.7-linux-x86_64.egg/gevent/greenlet.py", line 327, in run
    result = self._run(*self.args, **self.kwargs)
  File "/usr/local/lib/python2.7/dist-packages/gevent-1.0.2-py2.7-linux-x86_64.egg/gevent/server.py", line 102, in wrap_socket_and_handle
    ssl_socket = self.wrap_socket(client_socket, **self.ssl_args)
  File "/usr/local/lib/python2.7/dist-packages/gevent-1.0.2-py2.7-linux-x86_64.egg/gevent/_sslgte279.py", line 636, in wrap_socket
    ciphers=ciphers)
  File "/usr/local/lib/python2.7/dist-packages/gevent-1.0.2-py2.7-linux-x86_64.egg/gevent/_sslgte279.py", line 248, in __init__
    raise x
SSLEOFError: EOF occurred in violation of protocol (_ssl.c:590)
<Greenlet at 0x7fa4ef860d70: <bound method SocketIOServer.wrap_socket_and_handle of <SocketIOServer at 0x7fa4ef9e9590 fileno=8 address=0.0.0.0:8000>>(<socket at 0x7fa4ee41bd90 fileno=19 sock=172.31.16, ('2.239.61.150', 38717))> failed with SSLEOFError
```

**Comment by janxb** [comment]
I post you the same comment as i did in the Bug #729:

Maybe your problem is the same as some others which were posted here before.
If this could be the case, maybe you try the solution mentioned here: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=766867#15

Just patch the file mentioned in the last response from Frank Eular (no. 15) and tell us if this solved your problem. For me, it solved the SSL-problem after an installation on debian jessie.

**Comment by fakedfaker** [comment]
THX janxb!
solved my ssl Problem on Ajenti with Debian Jessie!
running great now !

**Comment by lukasmrtvy** [comment]
Well, i solved NameError: global name 'SSLContext' is not defined on fresh install of Xubuntu 15.04 64amd:

 `File "/usr/lib/python2.7/dist-packages/gevent/greenlet.py", line 327, 
in run
    result = self._run(*self.args, **self.kwargs)
  File "/usr/lib/python2.7/dist-packages/gevent/server.py", line 102, 
in wrap_socket_and_handle
    ssl_socket = self.wrap_socket(client_socket, **self.ssl_args)
  File "/usr/lib/python2.7/dist-packages/gevent/ssl.py", line 382, in 
wrap_socket
    ciphers=ciphers)
  File "/usr/lib/python2.7/dist-packages/gevent/ssl.py", line 84, in 
__init__
    ctx = SSLContext(ssl_version)
NameError: global name 'SSLContext' is not defined
<Greenlet at 0x7f7c2be9fd70: <bound method 
SocketIOServer.wrap_socket_and_handle of <SocketIOServer at 
0x7f7c2b83a0d0 fileno=6 address=10.192.254.16:1443>>(<socket at 
0x7f7c3623a950 fileno=10 sock=10.192.25, ('10.192.168.112', 54984))> 
failed with NameError`

with Frank Eular https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=766867#15 patch, but after this i have another error: SSLEOFError: EOF occurred in violation of protocol. I think its because of selfsigned cert?

`Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/gevent/greenlet.py", line 327, in run
    result = self._run(*self.args, **self.kwargs)
  File "/usr/lib/python2.7/dist-packages/gevent/server.py", line 102, in wrap_socket_and_handle
    ssl_socket = self.wrap_socket(client_socket, **self.ssl_args)
  File "/usr/lib/python2.7/dist-packages/gevent/ssl.py", line 382, in wrap_socket
    ciphers=ciphers)
  File "/usr/lib/python2.7/dist-packages/gevent/ssl.py", line 93, in __init__
    self.do_handshake()
  File "/usr/lib/python2.7/dist-packages/gevent/ssl.py", line 304, in do_handshake
    return self._sslobj.do_handshake()
SSLEOFError: EOF occurred in violation of protocol (_ssl.c:590)
<Greenlet at 0x7fcdda3ee230: <bound method SocketIOServer.wrap_socket_and_handle of <SocketIOServer at 0x7fcdda9d29d0 fileno=6 address=0.0.0.0:8000>>(<socket at 0x7fcdda3ecdd0 fileno=13 sock=10.0.0.41, ('10.0.0.34', 55474))> failed with SSLEOFError`

Any help here? Its easy to reproduce it. Thanks

**Comment by alessiodionisi** [comment]
Same error for me:

```
SSLEOFError: EOF occurred in violation of protocol
```

I've been using ajenti for months without ssl for this problem

**Comment by alessiodionisi** [comment]
Problem solved with gevent 1.1b3:

```
sudo easy_install -U gevent
```

**Comment by 1980Gandalf** [comment]
Hi.

I had same issue in centos 7, and I discovered disabling SSL, restarting ajenti, enabling SSL and restarting ajenti again solve the issue. I attach you a script that automates disable, restarting, enable and restarting of SSL and ajenti panel respectively.
[ajentipanelfix - copia.txt](https://github.com/ajenti/ajenti/files/4569493/ajentipanelfix.-.copia.txt)

And add it to cron at server reboots or image restores:
crontab -e
and add
@reboot sleep 20 && /usr/bin/ajentipanelfix.sh
I added a 20 seconds delay because I'm not sure if it should be:
@reboot  /usr/bin/ajentipanelfix.sh

I tried to update gevent library but panel stopped working because needed older version.

Regards.
